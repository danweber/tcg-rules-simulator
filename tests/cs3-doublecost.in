VERSION:2

When one of your Monster would evolve into a level 5 green Monster card, by suspending this placing 1 card from under one of your Tamers as the bottom evolution card of that Monster, reduce the evolution cost by 1

# [Your Turn] When one of your green Monster would be played, by placing 1 card from under one of your Tamers as the bottom evolution card of that Monster, reduce the play cost by 1.

FIELD1:FalseHit,CS3-Sorcery Tamer,CS3-Birdwatcher,CS3-02,CS3-03 Evo1,CS3-06 Evo2,CS3-06

HAND1:CS3-11 CS3-11
GAME:P1 T3 MAIN 10
DUMP

ECHO: ------ evo,  should interrupt
CMD:EVO CS3-11 Evo1
EFFECT-TREE
GET-CHOICES
DUMP

ECHO: -- don't pay, skips next question and evo for 4
STEP:TO:1
EFFECT-TREE
GET-CHOICES
DUMP


ECHO: ------ evo,  should interrupt
CMD:EVO CS3-11 Evo2
EFFECT-TREE
GET-CHOICES
DUMP

ECHO: -- choose to pay, the tamer suspends and then we're asked what to tuck
STEP:TO:0
EFFECT-TREE
GET-CHOICES
DUMP

ECHO: -- choose the featherlink, it's tucked and we evo for 3 instead of 4
STEP:TO:0
EFFECT-TREE
GET-CHOICES
DUMP
