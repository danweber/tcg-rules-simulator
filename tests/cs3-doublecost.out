FIELD1:FalseHit,CS3-Sorceryswan Tamer,CS3-Birdwatcherbill,CS3-Fairyfeatherlink,CS3-Chirpflare Evo1,CS3-Shieldwing Evo2,CS3-Shieldwing
FIELD2:
GAME:P1 T3 MAIN 10
ECHO: ------ evo,  should interrupt
EFFECT-TREE:DSE DEPTH 1 PROCESSING:normal evolve
EFFECT-TREE:INT DEPTH 1 Birdwatcher Bill
GET-CHOICES:Choose:
GET-CHOICES:Activate Birdwatcher Bill suspending
GET-CHOICES:Don't pay/activate
FIELD1:FalseHit,CS3-Sorceryswan Tamer,CS3-Birdwatcherbill,CS3-Fairyfeatherlink,CS3-Chirpflare Evo1,CS3-Shieldwing Evo2,CS3-Shieldwing
FIELD2:
GAME:P1 T3 MAIN 10
ECHO: -- don't pay, skips next question and evo for 4
EFFECT-TREE:NOTHING
GET-CHOICES:Choose:
FIELD1:FalseHit,CS3-Sorceryswan Tamer,CS3-Birdwatcherbill,CS3-Fairyfeatherlink,CS3-Chirpflare Evo1,CS3-Illumifalcon,CS3-Shieldwing Evo2,CS3-Shieldwing
FIELD2:
GAME:P1 T3 MAIN 6
ECHO: ------ evo,  should interrupt
EFFECT-TREE:DSE DEPTH 1 PROCESSING:normal evolve
EFFECT-TREE:INT DEPTH 1 Birdwatcher Bill
GET-CHOICES:Choose:
GET-CHOICES:Activate Birdwatcher Bill suspending
GET-CHOICES:Don't pay/activate
FIELD1:FalseHit,CS3-Sorceryswan Tamer,CS3-Birdwatcherbill,CS3-Fairyfeatherlink,CS3-Chirpflare Evo1,CS3-Illumifalcon,CS3-Shieldwing Evo2,CS3-Shieldwing
FIELD2:
GAME:P1 T3 MAIN 6
ECHO: -- choose to pay, the tamer suspends and then we're asked what to tuck
EFFECT-TREE:DSE DEPTH 1 PROCESSING:normal evolve
EFFECT-TREE:INT DEPTH 1 Birdwatcher Bill
GET-CHOICES:Choose 1 target for moving card:
GET-CHOICES:Target FairyFeatherlink BATTLE-CS3-02
GET-CHOICES:Target ChirpFlare BATTLE-CS3-03
FIELD1:FalseHit,CS3-Sorceryswan Tamer,CS3-Birdwatcherbill,CS3-Fairyfeatherlink,CS3-Chirpflare,REST Evo1,CS3-Illumifalcon,CS3-Shieldwing Evo2,CS3-Shieldwing
FIELD2:
GAME:P1 T3 MAIN 6
ECHO: -- choose the featherlink, it's tucked and we evo for 3 instead of 4
EFFECT-TREE:NOTHING
GET-CHOICES:Choose 1 target for moving card:
FIELD1:FalseHit,CS3-Sorceryswan Tamer,CS3-Birdwatcherbill,CS3-Chirpflare,REST Evo1,CS3-Illumifalcon,CS3-Shieldwing Evo2,CS3-Illumifalcon,CS3-Shieldwing,CS3-Fairyfeatherlink
FIELD2:
GAME:P1 T3 MAIN 3
